feat(argo-cd): Add support for custom deployment labels

akolacz opened the pull request:
<!--
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.
-->

This change enables custom deployment labels, which can be used to add Datadog labels for monitoring or for any other purpose.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

<!-- Changes are automatically published when merged to `main`. They are not published on branches. -->

yu-croco commented:
@akolacz Though you checked below, it's not handled. Please handle it.

>I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).

akolacz commented on charts/argo-cd/Chart.yaml:
```diff
@@ -3,7 +3,7 @@ appVersion: v3.0.5
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 8.0.16
+version: 8.1.0
```
Good to me, I will change to patch version instead

akolacz commented:
> @akolacz Though you checked below, it's not handled. Please handle it.
> 
> > I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).

I put it in changed and not in added, I change that

```
artifacthub.io/changes: |
    - kind: changed
      description: Add support for custom deployment labels
```

yu-croco reviewed (approved):
LGTM
